### DISCUSSION TITLE
Connection problem

### DISCUSSION THREAD

**Discussion by JonaszC9** [pymodbus-dev/pymodbus#1160] [Help] [answered]
Hi everyone, I'm trying to combine OpenCV and pymodbus in my project. This will be the detection of objects with a camera connected to the HMI panel. At first, I wanted to set up the connection properly, but I have a problem with that. In my case, the raspberry PI is the server and the laptop (later HMI) is the client. I was able to establish communication between devices with Modbus Client Simulator on laptop and Modbus Server Simulator on RPI, but when I try to run my simple code on RPI the connection is not working.

Python version: 3.9.2
Pymodbus version: 3.0.0

`from pymodbus.server import StartTcpServer
from pymodbus.datastore import ModbusSequentialDataBlock
from pymodbus.datastore import ModbusSlaveContext, ModbusServerContext
import logging

logging.basicConfig()
log = logging.getLogger()
log.setLevel(logging.DEBUG)

def run_server():
    store = ModbusSlaveContext(
        ir=ModbusSequentialDataBlock(30001, [100, 200]),
        zero_mode=True
    )
    context = ModbusServerContext(slaves=store, single=True)
    StartTcpServer(context=context, address=("localhost", 5020))

if __name__ == "__main__":
    run_server()`

What I receive in terminal is:
'DEBUG selector_events:59 Using selector: EpollSelector'

I tried to use examples from the site, they also don't work. Is there something wrong with my code or should I look elsewhere for the problem?

**Comment by janiversen** [comment, marked as answer]
I am not sure how I can help you, I have a couple of RPI4 working 24/7 with v3.0.0 as server and client. 

You need to do a couple of things:
- What do you run on the RPI ?
- Are you using external dongles, the builtin ethernet or the builtin wifi (depends on RPI model)
- Please activate debug log as described in the README and when opening an issue, do it on both server and client
Without these informations it is impossible to help you more.

**Reply by JonaszC9** [reply]
Thanks for the response. 

- I have RPI4 and run Debian 11.5
- I'm using just the builtin ethernet port
- I tried to, but when I run the code, this issue occurs: 
![fow](https://user-images.githubusercontent.com/117302058/200011027-3dbd236f-6083-4c5a-aa67-0917fe488d95.png)
`     from examples.helper import get_commandline
ModuleNotFoundError: No module named 'examples'`

Is this module from pymodbus? I couldn't find it anywhere.

**Reply by janiversen** [reply]
There are no module named examples....there are a DIRECTORY named examples, it is in the code base https://github.com/riptideio/pymodbus/tree/dev/examples

**Reply by JonaszC9** [reply]
I understand now the problem and where to look at. I'm just starting my RPi adventure. Not everything is clear to me yet. Thank you very much for your help :)
